### PR TITLE
Generate storage layout at pre-commit stage

### PR DESCRIPTION
**Pull request by huyhuynh3103**
### Description
- Run the command  `yarn plugin:storage-layout --override true` for generating storage layout command at pre-commit stage.
- The storage file will be extracted to `logs/storage_layout.txt`.

### CODE REVIEW

File: src/tasks/generate-storage-layout.ts [hunk 1]
```diff
@@ -65,6 +65,11 @@ const preprocessTable = (tableContent: string) => {
   return listItemOfTable;
 };
 
+function removeIdentifierSuffix(type: string) {
+  const identifierRegex = /\d+_(storage|memory|calldata)/g; // 12_memory 1123_storage
```
Comment (nxqbao) [inline src/tasks/generate-storage-layout.ts]: how about `t_contract(IRoninValidatorSet)11980`?

File: src/tasks/generate-storage-layout.ts [hunk 2]
```diff
@@ -65,6 +65,11 @@ const preprocessTable = (tableContent: string) => {
...
+  return type.replace(identifierRegex, '');
```
Comment (nxqbao) [inline src/tasks/generate-storage-layout.ts]: should keep the suffix identifier, but remove the id

idea:
```suggestion
  const identifierRegex = /(\d+)_(storage|memory|calldata)/g; // 12_memory 1123_storage
  return type.replace(identifierRegex, 'static_$2');
```

Comment (nxqbao) [reply src/tasks/generate-storage-layout.ts]: https://stackoverflow.com/a/3598167

**Review by nxqbao — APPROVED** [review]

**Review by nxqbao — APPROVED** [review]

**Review by nxqbao — APPROVED** [review]

**Comment by nxqbao** [comment]
Should change the dest file to `logs/storage_layout.txt`

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "build": "hardhat compile && tsc -p tsconfig.build.json",
     "plugin:size-contracts": "hardhat size-contracts",
     "plugin:upload-selectors": "hardhat upload-selectors",
-    "plugin:storage-layout": "hardhat check > layout/storage-layout.txt && yarn plugin:generate-storage-layout --source layout/storage-layout.txt",
+    "plugin:storage-layout": "hardhat check > logs/storage.txt && yarn plugin:generate-storage-layout --source logs/storage.txt",
```
Comment (nxqbao) [inline package.json:18]:
```suggestion
    "plugin:storage-layout": "hardhat check > logs/storage_layout.txt && yarn plugin:generate-storage-layout --source logs/storage_layout.txt",
```

Comment (huyhuynh3103) [reply package.json:18]: @nxqbao It is not the destination of the extracted file, but rather the temporary file generated by the hardhat-storage-layout package. This file will be deleted once the task is completed.

**Pull request merged**